src/Bongo/BongoClient.cs

```diff
@@ -3,113 +3,60 @@
 using System.Linq;
 using System.Net;
 using System.Threading.Tasks;
-using Bongo.Impala;
+using Akka.Actor;
+using Akka.Routing;
+using Bongo.Actors.BulkOperations;
+using Bongo.Actors.BulkOperations.Messages;
+using Bongo.Actors.Pools;
+using Bongo.Actors.Query;
+using Bongo.Actors.Query.Messages;
+using Bongo.Actors.Tables;
 
 namespace Bongo
 {
     public class BongoClient : IDisposable
     {
-        private readonly TimeSpan _timeout;
-        private readonly ImpalaClient _impalaClient;
+        private readonly ActorSystem _system;
+        private readonly IActorRef _bulkManager;
+        private readonly IActorRef _queryManager;
 
-        public BongoClient(IPEndPoint host, TimeSpan? timeout = null)
+        public BongoClient(List<IPEndPoint> impalaHosts, int hostPoolSize)
         {
-            _timeout = timeout.GetValueOrDefault(TimeSpan.FromSeconds(30));
-            _impalaClient = new ImpalaClient(host);
-        }
+            _system = ActorSystem.Create($"BongoClient-{Guid.NewGuid()}");
+            _system.ActorOf(Props.Create(() => new PoolManager(impalaHosts, hostPoolSize)), "poolManager");
+            _system.ActorOf(Props.Create(() => new TablesManager()), "tablesManager");
 
-        public Task<string> GetImpalaVersion()
-        {
-            return _impalaClient.GetImpalaVersion();
+            _queryManager = _system.ActorOf(Props.Create<QueryWorker>().WithRouter(new SmallestMailboxPool(5)), "queryManager");
+            _bulkManager = _system.ActorOf(Props.Create<BulkInsertWorker>().WithRouter(new SmallestMailboxPool(5)), "bulkInsertManager");
         }
 
-        public async Task CreateDatabase(string databaseName)
+        public async Task Insert<T>(List<T> items, bool upsert = false) where T : class
```

Comment:
Unless there's a specific need for List<>, you could support more types of collections with IEnumerable